OBAKitCore/Extensions/Collections.swift

```diff
@@ -26,4 +26,34 @@ extension Sequence {
     public func filter<T>(type: T.Type) -> [T] {
         return compactMap {$0 as? T}
     }
+
+    /// Sorts by key path.
+    func sorted<T: Comparable>(by keyPath: KeyPath<Element, T>) -> [Element] {
+        return sorted { a, b in
+            return a[keyPath: keyPath] < b[keyPath: keyPath]
+        }
+    }
+}
+
+extension Array {
+    /// Performs a binary search.
+    /// - important: The array must be sorted.
+    /// - parameter sortedBy: Which key the array is sorted by.
+    /// - parameter element: The element to look for.
+    /// - returns: The element and its index, if found. `nil` if not found.
+    func binarySearch<T: Comparable>(sortedBy keyPath: KeyPath<Element, T>, element: T) -> (index: Int, element: Element)? {
```

Comment:
nothing comparable exists in the stdlib?

Reply:
since i'm using generics and keypath here, it'll probably be more straightforward like this